src/components/UserSelector.tsx

```diff
@@ -2,24 +1,28 @@
+import React, { useEffect, useState } from 'react';
 import classNames from 'classnames';
-import { UserContext } from './UsersContext';
 import { User } from '../types/User';
+import { useDispatch, useSelector } from 'react-redux';
+import { setAuthor } from '../features/author';
+import { RootState } from '../app/store';
 
 type Props = {
   value: User | null;
   onChange: (user: User) => void;
 };
 
-export const UserSelector: React.FC<Props> = ({
-  // `value` and `onChange` are traditional names for the form field
-  // `selectedUser` represents what actually stored here
-  value: selectedUser,
-  onChange,
-}) => {
+export const UserSelector: React.FC<Props> = ({}) => {
   // `users` are loaded from the API, so for the performance reasons
   // we load them once in the `UsersContext` when the `App` is opened
   // and now we can easily reuse the `UserSelector` in any form
-  const users = useContext(UserContext);
+
+  const dispatch = useDispatch();
+  const users = useSelector((state: RootState) => state.users.items);
+
+  const author = useSelector(state => state.author);
```

Comment:
Consider specifying the `RootState` type in the `useSelector` hook to ensure type safety and avoid potential type inference issues.